aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -76,101 +78,233 @@ void filterEngineConfigs(
 
 struct CacheKey {
   ConvolutionParams params;
-  uint8_t input_alignment;
-  uint8_t weight_alignment;
-  uint8_t output_alignment;
+  cudnnBackendDescriptorType_t operation;
+  uint8_t x_alignment;
+  uint8_t w_alignment;
+  uint8_t y_alignment;
 };
 
 // FIXME: make this thread-safe by reusing the benchmark cache in Conv_v7.cpp
-std::unordered_map<CacheKey, cudnn_frontend::ManagedOpaqueDescriptor, ParamsHash<CacheKey>, ParamsEqual<CacheKey>> engine_cache;
+std::unordered_map<CacheKey, cudnn_frontend::ExecutionPlan, ParamsHash<CacheKey>, ParamsEqual<CacheKey>> engine_cache;
 
 }
 
-void raw_cudnn_convolution_forward_out(
-    const Tensor& output, const Tensor& input, const Tensor& weight,
-    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
-    bool benchmark, bool deterministic, bool allow_tf32)
-{
-  TORCH_CHECK(!benchmark, "not supported yet");
-  if (output.numel() == 0) {
-    return;
-  }
-
-  cudnnHandle_t handle = getCudnnHandle();
-
-  CacheKey key;
-  setConvolutionParams(&key.params, input, weight, padding, stride, dilation, groups, deterministic, allow_tf32);
-  key.input_alignment = getAlignment(input);
-  key.output_alignment = getAlignment(output);
-  key.weight_alignment = getAlignment(weight);
-
-  auto run = [&](cudnn_frontend::ManagedOpaqueDescriptor cfg) {
-    auto plan = cudnn_frontend::ExecutionPlanBuilder()
-        .setHandle(handle)
-        .setEngineConfig(cfg)
-        .build();
+void get_cachekey(CacheKey& key, const cudnnBackendDescriptorType_t operation, const Tensor& y, const Tensor& x, const Tensor& w, IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups, bool deterministic, bool allow_tf32) {
+   memset(&key, 0, sizeof(key));
```

Comment:
Why do we need this memset? Is there any unset field?

Reply:
This is needed to make using `ParamsHash` with `CacheKey` safe as the hashing and equality operations are blindly done on the bytes; we need to zero out padding bytes that are not guaranteed to be set. `setConvolutionParams` does something similar for this reason.